pkg/analyze/longhorn.go

```diff
@@ -67,3 +96,46 @@ func analyzeLonghornNodeSchedulable(node *longhornv1beta1.Node) *AnalyzeResult {
 
 	return result
 }
+
+func analyzeLonghornReplica(replica *longhornv1beta1.Replica) *AnalyzeResult {
+	result := &AnalyzeResult{
+		Title: fmt.Sprintf("Longhorn Replica: %s", replica.Name),
+	}
+
+	if replica.Spec.FailedAt != "" {
+		result.IsWarn = true
+		result.Message = fmt.Sprintf("Longhorn replica %s failed at %s", replica.Name, replica.Spec.FailedAt)
+		return result
+	}
+
+	desired := replica.Spec.InstanceSpec.DesireState
+	actual := replica.Status.InstanceStatus.CurrentState
+
+	if desired != actual {
+		result.IsWarn = true
+		result.Message = fmt.Sprintf("Longhorn replica %s current status %q, should be %q", replica.Name, actual, desired)
+		return result
+	}
+
+	result.IsPass = true
+	result.Message = fmt.Sprintf("Replica is %s", actual)
+
+	return result
+}
+
+func stripRedactedLines(yaml []byte) []byte {
```

Comment:
yaml Unmarshal fails on lines like `ip: ***HIDDEN***`

Reply:
seems like this function will be a bit over aggressive, but it'll still work for what we need